packages/containers/src/Action/Action.connect.js

```diff
@@ -18,15 +19,25 @@ export function mapStateToProps(state, ownProps) {
 	const props = {
 		renderers: getRenderers(renderers),
 	};
+	if (ownProps.overlayComponent) {
+		props.overlayComponent = (
+			<Inject component={ownProps.overlayComponent} {...ownProps.overlayComponentProps} />
+		);
+	}
+
```

Comment:
I'm not convince by this way to handle that in containers.
IMO, this should be handled in ActionButton.connect.js.
* the action settings define the action, accepting a `overlayComponent` which is a string : the component definition id.
* In the settings this id has a component definition.
* the ActionButton.connect.js `mapStateToProps` resolve the `props.overlayComponent, replacing it with the component injection and the props from settings.

I'm not a pro in CMF but it seems more CMF-like don't you think ?
@jmfrancois any thoughts ?